de.dlr.sc.virsat.swtbot.test/src/de/dlr/sc/virsat/swtbot/test/PowerSummaryTest.java

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package de.dlr.sc.virsat.swtbot.test;
+
+
+import static org.junit.Assert.assertEquals;
+
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTable;
+import org.eclipse.swtbot.swt.finder.widgets.SWTBotTreeItem;
+import org.junit.Before;
+import org.junit.Test;
+
+import de.dlr.sc.virsat.concept.unittest.util.ConceptXmiLoader;
+import de.dlr.sc.virsat.model.dvlm.concepts.Concept;
+import de.dlr.sc.virsat.model.extension.budget.power.model.PowerEquipment;
+import de.dlr.sc.virsat.model.extension.budget.power.model.PowerState;
+import de.dlr.sc.virsat.model.extension.budget.power.model.PowerSummary;
+import de.dlr.sc.virsat.model.extension.ps.model.ConfigurationTree;
+import de.dlr.sc.virsat.model.extension.ps.model.ElementConfiguration;
+
+public class PowerSummaryTest extends ASwtBotTestCase {
+	private SWTBotTreeItem repositoryNavigatorItem;
+	
+	private SWTBotTreeItem configurationTree;
+	private SWTBotTreeItem elementConfiguration1;
+	private SWTBotTreeItem elementConfiguration2;
+
+	private static final Double AOCS_POWER = 20.0;
+	private static final Double AOCS_DUTYCYCLE = 20.0;
+	
+	private static final Double DATA_HANDLING_POWER = 50.0;
+	private static final Double DATA_HANDLING_DUTYCYCLE = 50.0;
+
+	private SWTBotTreeItem powerEquipment1;
+	private SWTBotTreeItem powerEquipment2;
+	private SWTBotTreeItem powerSummary;
+
+	private Concept conceptPower;
+
+	
+	@Before
+	public void before() throws Exception {
+		super.before();
+		//create necessary items for the test
+		conceptPower = ConceptXmiLoader.loadConceptFromPlugin(de.dlr.sc.virsat.model.extension.budget.power.Activator.getPluginId() + "/concept/concept.xmi");
```

Comment:
We also had to a new concept in another test case and decided to add it in ASwtBotTestCase instead of the concrete test case. Not sure if it makes sense for this concept, because I don't know if there are other UseCases we could run SWT tests for with this concept.

Reply:
I would rather put the core concepts ( tests / core / ps ) into ATestCase but not this concept as it is really about some domain content... And what benefit would we have if it would be in the ATestCase?

Reply:
Ok then we leave it there